File: atcoder.jp/contests/abc170/tasks/abc170_b/main.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> range(int stop) {
  vector<int> ret(stop);
  iota(ret.begin(), ret.end(), 0);
  return ret;
}

template <typename Predicate, typename T>
bool some(Predicate pred, const vector<T> &vec) {
  return any_of(vec.begin(), vec.end(), pred);
}

int main() {
  int X, Y;
  cin >> X >> Y;

  bool ans =
      some([&](int crane) { return 2 * crane == 4 * X - Y; }, range(X + 1));

  cout << (ans ? "Yes" : "No") << endl;
}

File: atcoder.jp/contests/abc170/tasks/abc170_b/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];

const main = () => {
  const [X, Y] = input().split(" ").map(Number);

  const ans = [...Array(X + 1)].some((_, crane) => 2 * crane === 4 * X - Y);

  console.log(ans ? "Yes" : "No");
};

main();
